app/models/shipit/task_definition.rb

```diff
@@ -37,9 +38,14 @@ def as_json
         action: action,
         description: description,
         steps: steps,
+        variables: variables.map(&:to_h),
         checklist: checklist,
         allow_concurrency: allow_concurrency?,
       }
     end
+
+    def task_variables(config_variables)
+      config_variables.map(&VariableDefinition.method(:new))
```

Comment:
I kinda like it, and @hkdsun probably copied from my code somewhere else: https://github.com/Shopify/shipit-engine/blob/6e10e148c328f9c27a77c3b653234149f5e9a0ca/app/models/shipit/deploy_spec.rb#L66